### PR TITLE
General improvements to build.gradle

### PR DESCRIPTION
**Pull request by grimreaper**
- bump to gradle 3.4
- move common test tasks into withType
- move printing into common tasks
- stop printing on started tests (speeds up tests)
- enable buildscan and idea and tasks-tree as generally useful plugins
- bump pmd version
- bump fb-contrib version
- stop clean by default
- remove author lines

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -267,126 +262,82 @@ clean {
...
-    }
-
-    // Report each test as it is started.
-    beforeTest { descriptor ->
-    	logger.lifecycle("Running test: " + descriptor)
-    }
 }
 
 // Do the lot!
-task all(dependsOn: ['clean', 'build', 'slowtest', 'javadoc', 'allReports']) {
```
Comment (LegacyKing) [inline build.gradle]: Why are we removing clean across the board?

File: code/gradle/autobuild.gradle [hunk 2]
```diff
@@ -32,7 +30,7 @@ task copySite(type: Copy) {
     outputs.upToDateWhen { false }
     from "${projectDir}/code/gradle/site"
     into "${destAutobuildDir}"
-    expand(copyright: '2016', version: version, timestamp: buildTimestamp)
+    expand(copyright: '2017', version: version, timestamp: buildTimestamp)
```
Comment (LegacyKing) [inline code/gradle/autobuild.gradle]: Any reason we can't automate the current year?

Comment (grimreaper) [inline code/gradle/autobuild.gradle]: Need to figure out how. Its likely possible

File: build.gradle (hunk 1 above)
Comment (grimreaper) [inline build.gradle]: Gradle should work correctly without `clean` in the vast majority of cases. When it doesn't, its a bug. We already explicitly run `clean` in CI.

File: code/gradle/autobuild.gradle (hunk 2 above)
Comment (grimreaper) [inline code/gradle/autobuild.gradle]: done

**Comment by grimreaper** [comment]
> Why are we including a binary ,jar?
I'm not actually sure. It looks like we already included it, and I'm just updating it.
 gradle/wrapper/gradle-wrapper.jar        | Bin 54208 -> 54212 bytes

I suspect that most other uses of gradle don't include it, but I haven't had the time to investigate this properly yet.

**Comment by grimreaper** [comment]
going to split this up into separate PRs

**Comment by grimreaper** [comment]
ref #2607

**Pull request closed without merging**